From 3e9be8a78f2219f9c1605454ef9f46f491ae6c51 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?And=C3=AF=C3=AF?= <gnu_andrew@member.fsf.org>
Date: Fri, 14 Mar 2014 21:51:43 +0000
Subject: [PATCH 1/4] GraphicsMagick CFLAGS detection

Currently, dvdauthor fails to build against GraphicsMagick as it
doesn't retrieve the CFLAGS from pkg-config (see [0]). The attached
patch simplifies configure.ac, using the pkg-config macro to obtain
the settings for GraphicsMagick.

[0] https://bugs.gentoo.org/show_bug.cgi?id=459976
--
Andii :-)

(cherry picked from commit 5d038fa736a7e78887431fb8c7fe63c92b242dd1)
Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 configure.ac    | 16 +---------------
 src/Makefile.am |  2 +-
 2 files changed, 2 insertions(+), 16 deletions(-)

diff --git a/configure.ac b/configure.ac
index c43df92..52e3cb3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -59,21 +59,7 @@ if test -n "$MAGICKCONFIG"; then
 fi
 
 if test "$usemagick" != 1; then
-    AC_CHECK_PROGS(GMAGICKCONFIG, [GraphicsMagick-config])
-    if test -n "$GMAGICKCONFIG"; then
-        ac_save_CPPFLAGS="$CPPFLAGS"
-        ac_save_LIBS="$LIBS"
-        MAGICK_CPPFLAGS="`$GMAGICKCONFIG --cppflags`"
-        MAGICK_LIBS="`$GMAGICKCONFIG --ldflags` `$GMAGICKCONFIG --libs`"
-        CPPFLAGS="$CPPFLAGS $MAGICK_CPPFLAGS"
-        LIBS="$MAGICK_LIBS $LIBS"
-        AC_CHECK_FUNC(DispatchImage, usemagick=1)
-        CPPFLAGS="$ac_save_CPPFLAGS"
-        LIBS="$ac_save_LIBS"
-        if test "$usemagick" = 1; then
-            AC_DEFINE(HAVE_GMAGICK, 1, [Whether the GraphicsMagick libraries are available])
-        fi
-    fi
+    PKG_CHECK_MODULES([MAGICK], [GraphicsMagick], usemagick=1; [AC_DEFINE(HAVE_GMAGICK, 1, [whether the GraphicsMagick libraries are available])], [:])
 fi
 
 if test "$usemagick" != 1; then
diff --git a/src/Makefile.am b/src/Makefile.am
index c7f89a9..5ef90c3 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -8,7 +8,7 @@ endif
 nodist_bin_SCRIPTS = dvddirdel
 
 AM_CPPFLAGS = -DSYSCONFDIR="\"$(sysconfdir)\"" \
-    @XML_CPPFLAGS@ @MAGICK_CPPFLAGS@ @FREETYPE_CPPFLAGS@ @FRIBIDI_CFLAGS@ @FONTCONFIG_CFLAGS@
+    @XML_CPPFLAGS@ @MAGICK_CFLAGS@ @FREETYPE_CPPFLAGS@ @FRIBIDI_CFLAGS@ @FONTCONFIG_CFLAGS@
 AM_CFLAGS = -Wall
 
 dvdauthor_SOURCES = dvdauthor.c common.h dvdauthor.h da-internal.h \
-- 
1.9.1

